### PR TITLE
feat(server): defects endpoints

### PR DESCRIPTION
**Pull request by KarinKazarian**
#

## Changes this PR introduces

- update endpoint that allows you to update the status and description of a defect
- get all bike defects endpoint that returns a list of part defects related to each bike

## How to test this PR

- update a defect's status by sending a patch request
      - Endpoint: /defects
      - Sample Body: `{"_id": 123456, "status": "Pending"}`
      
- get all the bikes that have defective parts
       - Endpoint: /defects/bikes

## Notes for reviewers

- Make sure you add defects to the parts of a bikes before doing a get request

## Checklist

- [x] I renamed the title of my pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).

- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Comment by Mackbellemore** [comment]
Task linked: [CU-q53b27 Bike tab for quality management page](https://app.clickup.com/t/q53b27)

File: server/app/services/DefectService.ts [hunk 1]
```diff
@@ -20,6 +22,25 @@ export class DefectService {
     return this.defectRepo.getList();
   }
 
+  public async getBikeDefects(): Promise<unknown[]> {
+    const bikes = await this.bikeService.getBikes();
+    const bikeDefects: unknown[] = [];
```
Comment (gordonpn) [inline server/app/services/DefectService.ts]: Is `unknown` really the best solution here HAHAHA. Maybe we can define a quick interface to type this return/variable correctly?

File: server/app/services/DefectService.ts [hunk 2]
```diff
@@ -40,6 +61,20 @@ export class DefectService {
...
+
+    if (!defect) throw new NotFoundError(`Cannot find defect with ID ${body._id}`);
+
+    const updatedDefect = await this.defectRepo.update(body._id, {
+      status: body.status ? body.status : defect.status,
+      description: body.description ? body.description : defect.description,
+    } as IDefect);
+    return updatedDefect;
+  }
+
```
Comment (gordonpn) [inline server/app/services/DefectService.ts:78]: I can update a defect with a user-defined status, I think you just need to update the [DefectModel](https://github.com/Mackbellemore/soen-390-team07/blob/1d48c873c4fef9c34d0a5fe9a77ed88c08be0408/server/app/models/DefectModel.ts#L14) to only take an enum of strings.

**Review by gordonpn — COMMENTED** [review]
> Make sure you add defects to the parts of a bikes before doing a get request

Is there currently a way with the available APIs to add a defective part to a bike?

File: server/app/services/DefectService.ts (hunk 1 above)
Comment (KarinKazarian) [reply server/app/services/DefectService.ts]: Hahahhah yes we can define an interface. I'll add one, I just wasn't sure if making an interface just for the return type is worth it cuz I don't see where else this interface would be used.

**Comment by KarinKazarian** [comment]
> > Make sure you add defects to the parts of a bikes before doing a get request
> 
> Is there currently a way with the available APIs to add a defective part to a bike?

No, because bikes don't have defects directly related to them, only the parts have defects. You just need to add a defect for one of the parts of the bike that you are testing, and when you do a get for bike defects it will be in the list .

File: server/app/services/DefectService.ts (hunk 2 above)
Comment (KarinKazarian) [reply server/app/services/DefectService.ts:78]: Good catch!!

**Review by gordonpn — APPROVED** [review]

**Review by Mackbellemore — APPROVED** [review]

**Pull request merged**